server/src/main/java/org/opensearch/cluster/routing/RoutingPool.java

```diff
@@ -66,4 +66,8 @@ public static RoutingPool getIndexPool(IndexMetadata indexMetadata) {
         }
         return LOCAL_ONLY;
     }
+
+    public static boolean isShardAndNodePoolRemote(ShardRouting shardRouting, RoutingAllocation allocation, RoutingNode node) {
```

Comment:
Asks for 2 methods, `isNodePoolRemote` and `isShardPoolRemote` since these checks look independent from each other

Reply:
Sure. I can update this.